File: 이것이 취업을 위한 코딩 테스트다/BFS_DFS/재귀_팩토리얼/재귀_팩토리얼.cpp
#include <iostream>
using namespace std;

int factorial(int num) {
  if(num <= 1) {
    return 1;
  }

  return num * factorial(num - 1);
}

int main() {
  int value = factorial(5);
  cout << value << endl;

  int result = 1;
  int num = 5;

  for (int i = 1; i <= num; i++) {
    result *= i;
  }

  cout << result << endl;
}

File: 이것이 취업을 위한 코딩 테스트다/BFS_DFS/재귀_팩토리얼/재귀_팩토리얼.java
public class 재귀_팩토리얼 {
  public static void main(String[] args) {
    int num = 5;
    System.out.println(factorial_1(num));
    System.out.println(factorial_2(num));
  }
  
  public static int factorial_1(int num) {
    if (num <= 1) {
      return 1;
    }
  
    return num * factorial_1(num - 1);
  }
  
  public static int factorial_2(int num) {
    int result = 1;
  
    for (int i = 1; i <= num; i++) {
      result *= i;  
    }
  
    return result;
  }

}
